### PR TITLE
Base api for rest transition

### PR DESCRIPTION
**Pull request by DarylPenetrante**
## Description
- This PR introduces a new Base class for the Memsource REST api.
- Inheriting classes are the different resources (Jobs, Translation Memories, Analysis, etc.) and will be pushed in smaller PRs.

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -6,6 +6,13 @@ All notable changes to this project will be documented in this file. Dates are i
 Unreleased
 ==========
 
+[0.4.12] - 2020-06-17
```
Comment (ryanannatuvasquez) [inline CHANGELOG.rst:9]: I feel like this should be a new major version, transitioning to REST is kinda big step.

File: memsource/api_rest.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+            key: value for key, value in [
+                ('files', files), ('params', params), ('json', data), ('headers', self.headers)
+            ] if value is not None
+        }
+
+        # If it is successful, returns response json
+        return self._get_response(http_method, url, timeout=timeout, **arguments)
+
+    @staticmethod
+    def is_success(status_code: int):
```
Comment (jann-Briza) [inline memsource/api_rest.py:188]: [nit]
```diff
- def is_success(status_code: int):
+ def is_success(status_code: int) -> int:
```

File: memsource/api_rest.py [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+        except requests.exceptions.Timeout:
+            raise exceptions.MemsourceApiException(None, {
+                'errorCode': 'Internal',
+                'errorDescription': 'The request timed out, timeout is {}'.format(
+                    kwargs['timeout'] if 'timeout' in kwargs else 'default'),
+            }, self.last_url, self.last_params)
+        except requests.exceptions.ConnectionError:
+            raise exceptions.MemsourceApiException(None, {
+                'errorCode': 'Internal',
+                'errorDescription': 'Could not connect',
```
Comment (jann-Briza) [inline memsource/api_rest.py:141]: maybe it's still better if we print the acutal error?:
```diff
- 'errorDescription': 'Could not connect',
+ 'errorDescription': 'Raw response {}'.format(response.text),
```

File: README.rst [hunk 4]
```diff
@@ -7,6 +7,12 @@ Memsource API Wrap Library for Python
 
 This library require Python 3.5. If Python 3.5 is not installed on your system, I recommened to use `Pythonz <https://github.com/saghul/pythonz>`_
```
Comment (roniemartinez) [inline README.rst:8]: [nit] I think this is not needed anymore.

File: memsource/api_rest.py [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+        Arguments:
+        session -- The session object to be used by BaseApi
+        """
+        cls._session = session
+
+    def _get(
+            self,
+            path: str,
+            params: Dict[str, Any]={},
+            *,
```
Comment (roniemartinez) [inline memsource/api_rest.py:40]: Do we need this here?

File: README.rst (hunk 4 above)
Comment (DarylPenetrante) [reply README.rst:8]: I guess it's safe to leave it here. Not all user's of this library may have Python 3.5.

**Pull request merged**